include/seqan3/range/container/dynamic_bitset.hpp

```diff
@@ -0,0 +1,1837 @@
+        tmp.flip();
+        return tmp;
+    }
+
+    /*!\brief Performs binary shift left on the current object.
+     * \param[in] count Amount to shift to the left.
+     * \returns *this
+     *
+     * \details
+     *
+     * ### Example
+     *
+     * \include test/snippet/range/container/dynamic_bitset_shift_left_inplace.cpp
+     *
+     * ### Exception
+     *
+     * No-throw guarantee.
+     *
+     * ### Thread-safety
+     *
+     * Not thread-safe.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     */
+    constexpr dynamic_bitset & operator<<=(size_t const count) noexcept
+    {
+        assert(count > 0);
+        assert(count < size());
+        uint8_t offset = 58 - size() + count;
+        data_.bits <<= offset;
+        data_.bits >>= offset - count;
+        return *this;
+    }
+
+    /*!\brief Performs binary shift right on the current object.
+     * \param[in] count Amount to shift to the right.
+     * \returns *this
+     *
+     * \details
+     *
+     * ### Example
+     *
+     * \include test/snippet/range/container/dynamic_bitset_shift_right_inplace.cpp
+     *
+     * ### Exception
+     *
+     * No-throw guarantee.
+     *
+     * ### Thread-safety
+     *
+     * Not thread-safe.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     */
+    constexpr dynamic_bitset & operator>>=(size_t const count) noexcept
+    {
+        assert(count > 0);
+        assert(count < size());
+        data_.bits >>= count;
+        return *this;
+    }
+
+    /*!\brief Performs binary shift right.
+     * \param[in] count Amount to shift to the right.
+     * \returns Copy of *this with bits shifted count to the right.
+     *
+     * \details
+     *
+     * ### Example
+     *
+     * \include test/snippet/range/container/dynamic_bitset_shift_right_copy.cpp
+     *
+     * ### Exception
+     *
+     * No-throw guarantee.
+     *
+     * ### Thread-safety
+     *
+     * Not thread-safe.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     */
+    constexpr dynamic_bitset operator>>(size_t const count) const noexcept
+    {
+        assert(count > 0);
+        assert(count < size());
+        dynamic_bitset tmp{*this};
+        tmp >>= count;
+        return tmp;
+    }
+
+    /*!\brief Performs binary shift left.
+     * \param[in] count Amount to shift to the left.
+     * \returns Copy of *this with bits shifted count to the left.
+     *
+     * \details
+     *
+     * ### Example
+     *
+     * \include test/snippet/range/container/dynamic_bitset_shift_left_copy.cpp
+     *
+     * ### Exception
+     *
+     * No-throw guarantee.
+     *
+     * ### Thread-safety
+     *
+     * Not thread-safe.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     */
+    constexpr dynamic_bitset operator<<(size_t const count) const noexcept
+    {
+        assert(count > 0);
+        assert(count < size());
+        dynamic_bitset tmp{*this};
+        tmp <<= count;
```

Comment:
overflow check?

Reply:
Mhm, do we want an assert or something?
Behaviour is:
If you do `<<=` greater than capacity, then the data is 0, but the size bits are not affected, i.e. same effect as reset.
If you shift more than 63, you get the standard complaint:
```
dynamic_bitset.hpp:680:20: error: right operand of shift expression '(4288 << 64)' is >= than the precision of the left operand [-fpermissive]
  680 |         data_.bits <<= count;
```

Reply:
But, we do assert in debug mode:
```cpp
        assert(count > 0);
        assert(count < size());
```